module/enrichers.mjs

```diff
@@ -363,40 +399,62 @@ function createRollLink(label, dataset) {
+  const target = event.target.closest(".roll-link") ?? event.target.closest('[data-action="rollRequest"]');
   if ( !target ) return;
   event.stopPropagation();
 
   const { type, ability, skill, tool, dc } = target.dataset;
   const options = { event };
   if ( dc ) options.targetValue = dc;
 
-  // Fetch the actor that should perform the roll
-  let actor;
-  const speaker = ChatMessage.implementation.getSpeaker();
-  if ( speaker.token ) actor = game.actors.tokens[speaker.token];
-  actor ??= game.actors.get(speaker.actor);
-  if ( !actor && (type !== "damage") ) {
-    ui.notifications.warn(game.i18n.localize("EDITOR.DND5E.Inline.NoActorWarning"));
-    return;
+  const action = event.target.closest("a")?.dataset.action ?? "roll";
+
+  // Direct roll
+  if ( (action === "roll") || !game.user.isGM ) {
+    // Fetch the actor that should perform the roll
+    let actor;
+    const speaker = ChatMessage.implementation.getSpeaker();
+    if ( speaker.token ) actor = game.actors.tokens[speaker.token];
+    actor ??= game.actors.get(speaker.actor);
+    if ( !actor && (type !== "damage") ) {
+      ui.notifications.warn(game.i18n.localize("EDITOR.DND5E.Inline.NoActorWarning"));
+      return;
+    }
+
+    switch ( type ) {
+      case "check":
+        return actor.rollAbilityTest(ability, options);
+      case "damage":
+        return rollDamage(event, speaker);
+      case "save":
+        return actor.rollAbilitySave(ability, options);
+      case "skill":
+        if ( ability ) options.ability = ability;
+        return actor.rollSkill(skill, options);
+      case "tool":
+        options.ability = ability;
+        return actor.rollToolCheck(tool, options);
+      default:
+        return console.warn(`DnD5e | Unknown roll type ${type} provided.`);
+    }
   }
 
-  switch ( type ) {
-    case "check":
-      return actor.rollAbilityTest(ability, options);
-    case "damage":
-      return rollDamage(event, speaker);
-    case "save":
-      return actor.rollAbilitySave(ability, options);
-    case "skill":
-      if ( ability ) options.ability = ability;
-      return actor.rollSkill(skill, options);
-    case "tool":
-      options.ability = ability;
-      return actor.rollToolCheck(tool, options);
-    default:
-      return console.warn(`DnD5e | Unknown roll type ${type} provided.`);
+  // Roll request
+  else {
+    const MessageClass = getDocumentClass("ChatMessage");
+    const chatData = {
+      user: game.user.id,
+      type: CONST.CHAT_MESSAGE_TYPES.OTHER,
+      content: await renderTemplate("systems/dnd5e/templates/chat/request-card.hbs", {
+        buttonLabel: createRollLabel({ ...target.dataset, format: "short" }),
+        dataset: { ...target.dataset, action: "rollRequest" }
+      }),
+      flavor: game.i18n.localize("EDITOR.DND5E.Inline.RollRequest"),
+      speaker: MessageClass.getSpeaker({user: game.user})
+    };
+    MessageClass.applyRollMode(chatData, game.settings.get("core", "rollMode"));
```

Comment:
It looks like it is only applied automatically when the type of chat messages is `roll`.

I guess there is the question of whether respecting roll mode setting is useful here, since only GMs can click these buttons and they would only be able to restrict the messages to other GMs.